packages/web/src/application/test/zoom/zoom-provider.test.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { ReactNode } from 'react';
+import { MemoryRouter } from 'react-router-dom';
+import { render, screen, waitFor } from '@testing-library/react';
+import { User } from '@chess-tent/models';
+import application from '@application';
+import { users, zoom } from '../fixtures';
+
+const { components } = application;
+
+interface RenderOptions {
+  children: ReactNode;
+  user: User;
+  meetingNumber?: string;
+  queryCode?: boolean;
+}
+
+const renderInsideProvider = ({
+  children,
+  user,
+  meetingNumber,
+  queryCode,
+}: RenderOptions) => {
+  const { ZoomProvider } = components;
+
+  render(
+    <MemoryRouter
+      initialEntries={[
+        { pathname: '/', search: queryCode ? '?code=testcode' : undefined },
+      ]}
+    >
+      <ZoomProvider
+        redirectUri={zoom.redirectUri}
+        user={user}
+        meetingNumber={meetingNumber}
+      >
+        {children}
+      </ZoomProvider>
+    </MemoryRouter>,
+  );
+};
+
+describe('Zoom Provider', () => {
+  const coach: User = users.find(user => user.coach)!;
+  const student: User = users.find(user => !user.coach)!;
+
+  test('Student should have only password input', async () => {
+    const { ZoomGuestControl } = components;
+
+    renderInsideProvider({
+      children: <ZoomGuestControl />,
+      user: student,
+      meetingNumber: zoom.meetingNumber,
+    });
+
+    const passwordInput = screen.queryByPlaceholderText(
+      'Meeting password (if any)',
+    );
+    const meetingNumberInput = screen.queryByPlaceholderText('Meeting number');
+
+    await waitFor(() => expect(passwordInput).toBeTruthy());
+    await waitFor(() => expect(meetingNumberInput).toBeFalsy());
+  });
+
+  test('Coach should have authorize button only when no authCode', async () => {
+    const { ZoomHostControl } = components;
+
+    renderInsideProvider({
+      children: <ZoomHostControl />,
+      user: coach,
+    });
+
+    const authButton = screen.getByRole('button', {
```

Comment:
Always use `getByText` when possible.

Reply:
Why?

Reply:
In general it's better to do things less specifically.

When you get element by role and then by name you automatically have more specific logic which is irrelevant in this case.

It's not important that the role of element is button.

In the end you'll have a simpler/more readable function call.